common/thorhelper/thorread.cpp

```diff
@@ -556,11 +703,458 @@ size32_t BinaryDiskRowReader::getFixedDiskRecordSize()
+    RowFilter projectedFilter;
+    const RtlRecord * projectedRecord = nullptr;
+};
+
+class CNullNestedRowIterator : public CSimpleInterfaceOf<IDynamicRowIterator>
+{
+public:
+    virtual bool first() override { return false; }
+    virtual bool next() override { return false; }
+    virtual bool isValid() override { return false; }
+    virtual IDynamicFieldValueFetcher &query() override
+    {
+        throwUnexpected();
+    }
+} nullNestedRowIterator;
+
+//---------------------------------------------------------------------------------------------------------------------
+
+class CsvDiskRowReader : public ExternalFormatDiskRowReader
+{
+private:
+    class CFieldFetcher : public CSimpleInterfaceOf<IDynamicFieldValueFetcher>
+    {
+        CSVSplitter &csvSplitter;
+        unsigned numInputFields;
+    public:
+        CFieldFetcher(CSVSplitter &_csvSplitter, unsigned _numInputFields) : csvSplitter(_csvSplitter), numInputFields(_numInputFields)
+        {
+        }
+        virtual const byte *queryValue(unsigned fieldNum, size_t &sz) const override
+        {
+            dbgassertex(fieldNum < numInputFields);
+            sz = csvSplitter.queryLengths()[fieldNum];
+            return csvSplitter.queryData()[fieldNum];
+        }
+        virtual IDynamicRowIterator *getNestedIterator(unsigned fieldNum) const override
+        {
+            return LINK(&nullNestedRowIterator);
+        }
+        virtual size_t getSize(unsigned fieldNum) const override { throwUnexpected(); }
+        virtual size32_t getRecordSize() const override { throwUnexpected(); }
+    };
+
+public:
+    CsvDiskRowReader(IDiskReadMapping * _mapping);
+
+    virtual const void *nextRow() override;
+    virtual const void *nextRow(size32_t & resultSize) override;
+    virtual void stop() override;
+
+    virtual bool matches(const char * format, bool streamRemote, IDiskReadMapping * otherMapping) override;
+
+protected:
+    virtual bool setInputFile(IFile * inputFile, const char * _logicalFilename, unsigned _partNumber, offset_t _baseOffset, const IPropertyTree * meta, const FieldFilterArray & expectedFilter) override;
+
+    void processOption(CSVSplitter::MatchItem element, const IPropertyTree & config, const char * option, const char * dft, const char * dft2 = nullptr);
+
+    inline bool fieldFilterMatchProjected(const void * buffer)
+    {
+        if (projectedFilter.numFilterFields())
+        {
+            unsigned numOffsets = projectedRecord->getNumVarFields() + 1;
+            size_t * variableOffsets = (size_t *)alloca(numOffsets * sizeof(size_t));
+            RtlRow row(*projectedRecord, nullptr, numOffsets, variableOffsets);
+            row.setRow(buffer, 0);  // Use lazy offset calculation
+            return projectedFilter.matches(row);
+        }
+        else
+            return true;
+    }
+
+    size32_t getFixedDiskRecordSize();
+
+protected:
+    constexpr static unsigned defaultMaxCsvRowSize = 10; // MB
```

Comment:
Might be inclined to call the variable defaultMaxCsvRowSizeMB

Reply:
I agree.  I think this code was copied.